johnnyfive/confluence.py

```diff
@@ -0,0 +1,274 @@
+
+        Either for curiosity or for modification, get the page contents, which
+        live in the `body.storage` portion of the `get_page_by_id` response.
+
+        Returns
+        -------
+        `str`
+            The HTML-ish body of the confluence page.
+        """
+        contents = safe_confluence_connect(self.instance.get_page_by_id,
+                                          self.page_id,
+                                          expand='body.storage')
+        # Extract the contents from the return object
+        return contents['body']['storage']['value']
+
+    def smite(self):
+        """smite Kill with extreme prejudice
+
+        Remove the Confluence page and update the instance metadata to reflect
+        the new state.
+        """
+        safe_confluence_connect(self.instance.remove_page, self.page_id)
+        self._set_metadata()
+
+    def update_contents(self, body):
+        """update_contents Update the contents of the Confluence page
+
+        Update the page by replacing the existing content with new.  The idea
+        for this method is to be used in concert with `get_page_contents` to
+        obtain a page, modify it, then replace it.
+
+        Parameters
+        ----------
+        body : `str`
+            The new page contents to upload to Confluence.
+
+        Returns
+        -------
+        _type_
+            _description_
+        """
+        safe_confluence_connect(self.instance.update_page, self.page_id,
+                                self.title, body)
+
+    def _set_metadata(self):
+        """_set_metadata Set the various instance metadata
+
+        Especially after a page is created or deleted, this method updates the
+        various instance attributes to keep current.
+        """
+        self.exists = safe_confluence_connect(self.instance.page_exists,
+                                              self.space, self.title)
+
+        # Page-Specific Information
+        self.page_id = None if not self.exists else \
+            safe_confluence_connect(self.instance.get_page_id,
+                                    self.space, self.title)
+        self.attachment_url = None if not self.exists else \
+            f"{self.instance.url}download/attachments/{self.page_id}/"
+
+
+# Internal Functions =========================================================#
+def setup_confluence():
+    """setup_confluence Set up the Confluence class instance
+
+    Reads in the confluence.conf configuration file, which contains the URL,
+    username, and password.  Also contained in the configuration file are
+    the Confluence space and page title into which the updated table will be
+    placed.
+
+    Returns
+    -------
+    confluence : `atlassian.Confluence`
+        Confluence class, initialized with credentials
+    """
+    # Read the setup
+    setup = utils.read_ligmos_conffiles('confluenceSetup')
+
+    # Return the Confluence instance
+    return Confluence( url=setup.host,
```

Comment:
At some point, Atlassian might force OAuth for this, too - probably when we get herded into Confluence cloud.  At long as we run the server instance this will likely still be just fine, but those days are already numbered.  Might be worth revisiting after this is merged down.

Reply:
We cannot use OAuth with our current Confluence version (7.3.5) -- it requires >=7.9.

The outlines of how this will work has been added as a comment in this method.